### ISSUE TITLE
Installation issue: bufr with gfortran if `fc` is not just `gfortran`

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack#42]
### Steps to reproduce the issue

The following code in the bufr `package.py` only works if the Fortran compiler `fc` is just `gfortran`:
```
        # Bufr inserts a path into source code which may be longer than 132
        if 'gfortran' in fc:
            flags.append('-ffree-line-length-none')
```
However, depending on the system and compiler config, `fc` could also be `/some/path/to/gfortran` or `gfortran-11` etc.

### Error message

Compilation fails with an error that a line gets truncated (the line where the preprocessor generates an include statement with an absolute path).

### Information on your system

macOS with llvm-clang@13.0.0 and gfortran-11.2.0

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by kgerheiser** [comment]
Is there actually an issue here? `'gfortran' in '/usr/bin/gfortran-11' == True`

**Comment by climbfuji** [comment]
Not anymore ... I fixed something and put the correct code in the issue:

https://github.com/NOAA-EMC/spack/commit/d38fadc60af0f26230b37efb2189e6a6099f2361

This was fixed by #43 :-)
